packages/phone-number-privacy/combiner/src/domain/endpoints/disable/action.ts

```diff
@@ -27,25 +34,4 @@ export class DomainDisableAction extends CombineAction<DisableDomainRequest> {
       session.logger
     )
   }
-
-  protected async receiveSuccess(
-    signerResponse: FetchResponse,
-    url: string,
-    session: Session<DisableDomainRequest>
-  ): Promise<DisableDomainResponse> {
-    const res = await super.receiveSuccess(signerResponse, url, session)
-    if (this.checkThresholdDisabled(session)) {
```

Comment:
I'm a little confused about this. It looks to me like what will happen if a threshold of signers successfully disable the domain is that the the threshold-state service will either return a zeroed out domain state or throw "Insufficient number of signer responses. Domain may be disabled", both of which will cause us to send a failure response to the client even though the domain was actually disabled. I'm okay with not aborting the requests when a threshold is reached, but I do think we should return 200 iff a threshold of signers disable the domain rather than leaving that determination to the client. There's a good chance I'm misunderstanding so happy to chat through this once you're back if I'm not making sense :)

Reply:
I'm not totally sure what you mean: if a threshold of signers disable, we do return a 200 & the zeroed out domain state

> both of which will cause us to send a failure response to the client

In the first case (zeroed out domain state, i.e. domain is disabled), we return a 200 + this state from the combiner. Or do you mean something different? We are now just using the `state.disabled` field returned from the signer as opposed to only the success field (to keep this more in line with the other domain logic).

As for my PR description note: this more references the case that you mention about sending failure though the domain was actually disabled. (I think the initial Math.max logic wasn't strictly correct, and since we had this implemented differently in the `DomainThresholdStateService`, that logic was effectively the "disabling" logic when trying to get signatures from a disabled domain). We still have some weirdnesses I think with this indeterminate state; we could change this by taking the min (instead of the max), something that logically translates to:
```
const numDisabled = // calculate differently than currently, i.e. not total states - # enabled, instead just # disabled
if (signersLength - numDisabled < threshold || numDisabled >= threshold) {// return disabled domain state} 
```
This makes it easier to disable states and means that a malicious min(threshold, totalSigners - threshold) could disable a domain, but at least it would be unambiguous about what's going on.

May be best to just look at/discuss this in sync!

Reply:
> In the first case (zeroed out domain state, i.e. domain is disabled), we return a 200 + this state from the combiner.
Agreed, I must've misread this yesterday. 

> const numDisabled = // calculate differently than currently, i.e. not total states - # enabled, instead just # disabled
I'm pretty sure these two ways of calculating are equivalent, right? Both result in the number of disabled domain states received by the combiner. The only reason it's written this way currently is to simplify the code (since we use the enabledDomainStates array later). 

> This makes it easier to disable states and means that a malicious min(threshold, totalSigners - threshold) could disable a domain
A domain state is only disabled when it's impossible to reach a threshold of signers who still have it enabled (you can ping the signers directly without going through the combiner). If we have a system with a 2/6 threshold and 3 signers successfully disable the domain while the other 3 do not, the domain is not disabled because 3 enabled signers can still be reached. Yet, numDisabled >= threshold so we would incorrectly return a success given the logic above. I'm having trouble thinking of a scenario when the original logic (copied below) doesn't work, but maybe I'm missing something? 
isDisabled = numDisabled >= Math.max(threshold, this.signers.length - threshold + 1)

>

Reply:
> I'm pretty sure these two ways of calculating are equivalent, right? Both result in the number of disabled domain states received by the combiner.

Oop sorry for the confusion I was super unclear on this above; I added a note to self about this bc this only applies if we do this check before the threshold check (at which point we aren't sure that we have enough signers anyways, so we need to check the #disabled absolutely instead of based on the length at all (which we don't know is >= threshold)).

And yes agreed about the logic not currently being different than before (except for the early abort & one small bug that I will fix -- wasn't checking the disabled state return value); we can change this slightly/make it more explicit by changing where the actual disable domain check happens (i.e. before the threshold check).
--> double check, is it the same for T < N/2??
    - What happens before this PR for T <= N/2:
         call to disable endpoint -> 

Questions we should discuss that I think make it clearer how we want to implement this section:
- Is it worse to have signers maliciously disable a should-be-enabled domain or maliciously enable a should-be-disabled domain?
- Related to the first question, what should take precedence, enabling/disabling? If T <= N/2, we can have a threshold of both enable/disabled simultaneously. We currently prefer the responses that are enabled (i.e. even if a threshold has disabled it, a simultaneous threshold that enables it wins) -- is this desired? Should we actually just prevent ODIS this from being used with T <= N/2 entirely? May make sense to double check with Victor on this.
    - I think this preference has been confusing to me since it feels inconsistent (even if it isn't strictly so?): if T > N/2, then <T signers can effectively disable (i.e. we practically prefer disabling vs. enabling). However, if T <= N/2, then we need >T signers to disable (otherwise we prefer enabling in practice). I think a way of standardizing this behavior is we either ban the latter case (T <= N/2) OR in this case, we also prefer disabling if that hits threshold. This ofc depends on the first question (what tradeoff is preferable for malicious signers). Does this make sense? lol I swear every time I come back to the disabled domain logic stuff my brain breaks a bit haha :( Looking forward to chatting

As far as aborting early vs. not & returning the state or not, IMO it's just nicer/a bit more understandable to me to follow the same structure & calculation in each of the domain endpoints, and to really hand off all domain threshold logic to the DomainThresholdService. Missed this before, but there are actually some (weird) cases where we don't succeed in disabling but still get the state back, and can choose to return the domainState on error in those cases if we want to.

> and it seems like there's some complexity in adapting the domain threshold-service to intuitively accept DisableDomainResponses
I don't think there's more complexity than already exists, given that the other endpoints rely on disabled/enabled states too?

Reply:
decision here post talk: leave the logic the way it is for now and add a comment explaining this